services/queue/src/api.js

```diff
@@ -1036,12 +1028,7 @@ builder.declare({
     );
   }
 
-  const worker = await this.Worker.load({
-    provisionerId: task.provisionerId,
-    workerType: task.workerType,
-    workerGroup,
-    workerId,
-  }, true);
+  const worker = await Worker.get(this.db, null, null, workerGroup, workerId, new Date());
```

Comment:
`provisionerId` and `workerType` are both specified in the old query but they are set to `null` here. Should they be set?

Reply:
They should -- workerGroup+workerId is only defined to be unique within the context of a workerPool.  Also, omitting those parameters makes this a table scan.